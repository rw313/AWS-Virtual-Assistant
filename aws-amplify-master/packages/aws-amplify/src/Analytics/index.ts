/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import AnalyticsClass from './Analytics';

import {
    ConsoleLogger as Logger,
    Hub
} from '../Common';

const logger = new Logger('Analytics');

let _instance: AnalyticsClass = null;

if (!_instance) {
    logger.debug('Create Analytics Instance');
    _instance = new AnalyticsClass(null);
}

const Analytics = _instance;
export default Analytics;

Analytics.onHubCapsule = (capsule) => {
    const { channel, payload, source } = capsule;
    logger.debug('on hub capsule ' + channel, payload);

    switch(channel) {
        case 'auth':
            authEvent(payload);
            break;
        case 'storage':
            storageEvent(payload);
            break;
    }
};

const storageEvent = (payload) => {
    const { attrs, metrics } = payload;
    if (!attrs) return;

    Analytics.record('Storage', attrs, metrics);
};

const authEvent = (payload) => {
    const { event } = payload;
    if (!event) { return; }

    switch(event) {
        case 'signIn':
            Analytics.restart();
            Analytics.record('_userauth.sign_in');
            break;
        case 'signUp':
            Analytics.record('_userauth.sign_up');
            break;
        case 'signOut':
            Analytics.restart();
            break;
        case 'signIn_failure':
            Analytics.record('_userauth.auth_fail');
            break;
    }
};

Hub.listen('auth', Analytics);
Hub.listen('storage', Analytics);
