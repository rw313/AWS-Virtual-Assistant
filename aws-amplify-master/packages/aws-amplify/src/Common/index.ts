/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { AWS } from './Facet';
import { ConsoleLogger as Logger } from './Logger';

export * from './Facet';
export { default as ClientDevice } from './ClientDevice';
export * from './Logger';
export * from './Errors';
export { default as Hub } from './Hub';
export { default as JS } from './JS';
export { default as Signer } from './Signer';

import Platform from './Platform';
export const Constants = {
    'userAgent': Platform.userAgent
};

const logger = new Logger('Common');

if (AWS['util']) {
    AWS['util'].userAgent = () => {
        return Constants.userAgent;
    };
} else if (AWS.config) {
    AWS.config.update({'customUserAgent': Constants.userAgent});
} else {
    logger.warn('No AWS.config');
}
