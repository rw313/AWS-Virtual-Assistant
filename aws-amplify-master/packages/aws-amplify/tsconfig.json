//WARNING: If you are manually specifying files to compile then the tsconfig.json is completely ignored, you must use command line flags
{
    "compilerOptions": {
        "outDir": "./lib/",
        "target": "es5",
        "noImplicitAny": false,
        "lib": ["es5", "es2015","dom"],
        "sourceMap": true,
        "module": "commonjs",
        "moduleResolution": "node",
        "allowJs": false,
        "declaration": true
    },
    "include": ["src/**/*"]
 }