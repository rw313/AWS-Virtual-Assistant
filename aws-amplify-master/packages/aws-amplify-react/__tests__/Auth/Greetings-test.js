/*
jest.mock('aws-sdk-mobile-analytics', () => {
    const Manager = () => {}

    Manager.prototype.recordEvent = () => {

    }

    Manager.prototype.recordMonetizationEvent = () => {

    }

    var ret =  {
        Manager: Manager
    }
    return ret;
});

jest.mock('aws-sdk/clients/pinpoint', () => {
    const Pinpoint = () => {
        var pinpoint = null;
        return pinpoint;
    }

    Pinpoint.prototype.updateEndpoint = (params, callback) => {
        callback(null, 'data');
    }

    return Pinpoint;
});
*/

import Greetings from '../../src/Auth/Greetings';
import React from 'react';
import AmplifyTheme from '../../src/AmplifyTheme';
import AuthPiece from '../../src/Auth/AuthPiece';
import { Header, Footer, InputRow, ButtonRow } from '../../src/AmplifyUI';
import { Auth } from 'aws-amplify';

const acceptedStates = [
    'signedIn'
];

const deniedStates = [
    'signIn',  
    'signedUp', 
    'signedOut',
    'forgotPassword',
    'signUp',
    'confirmSignIn',
    'confirmSignUp',
    'verifyContact'
];

describe('signUp', () => {
    describe('normal case', () => {
        test('render correctly with authState signedIn', () => {
            const wrapper = shallow(<Greetings/>);
            for (var i = 0; i < acceptedStates.length; i += 1){
                wrapper.setProps({
                    authState: acceptedStates[i],
                    theme: AmplifyTheme
                });
                expect(wrapper).toMatchSnapshot();
            }
        });

        test('checkUser when component mount', async () => {
            const spyon = jest.spyOn(Auth, 'currentAuthenticatedUser')
                .mockImplementationOnce(() => {
                    return new Promise((res, rej) => {
                        res('user');
                    });
                });
            const wrapper = mount(<Greetings />);
            //wrapper.setState({authState: 'signedIn'});

            
        })
    });

   
    test('render corrently with other authStates', () => {
        const wrapper = shallow(<Greetings/>);
        
        for (var i = 0; i < deniedStates.length; i += 1){
            wrapper.setProps({
                authState: deniedStates[i],
                theme: AmplifyTheme
            });

            expect(wrapper).toMatchSnapshot();
        }
    });
});