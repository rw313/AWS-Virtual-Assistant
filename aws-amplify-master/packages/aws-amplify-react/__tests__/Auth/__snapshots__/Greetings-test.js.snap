// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp normal case render correctly with authState signedIn 1`] = `""`;

exports[`signUp render corrently with other authStates 1`] = `""`;

exports[`signUp render corrently with other authStates 2`] = `""`;

exports[`signUp render corrently with other authStates 3`] = `""`;

exports[`signUp render corrently with other authStates 4`] = `""`;

exports[`signUp render corrently with other authStates 5`] = `""`;

exports[`signUp render corrently with other authStates 6`] = `""`;

exports[`signUp render corrently with other authStates 7`] = `""`;

exports[`signUp render corrently with other authStates 8`] = `""`;
